### PR TITLE
Don't inject the eager extension service if it's disabled

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/api-platform/core/pull/822#discussion_r86358789
| License       | MIT
| Doc PR        | na

And now the 0.001 % are back \o/

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DoctrineQueryExtensionPass.php [hunk 1]
```diff
@@ -54,6 +54,12 @@ private function findSortedServices(ContainerBuilder $container, $tag)
     {
         $extensions = [];
         foreach ($container->findTaggedServiceIds($tag) as $serviceId => $tags) {
+            if ('api_platform.doctrine.orm.query_extension.eager_loading' === $serviceId
```
Comment (meyerbaptiste) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/DoctrineQueryExtensionPass.php]: IMO it should be done in the `ApiPlatformExtension` class with the `ContainerBuilder::removeDefinition()` method.

File: tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php [hunk 2]
```diff
@@ -390,6 +390,8 @@ private function getContainerBuilderProphecy()
 
         $containerBuilderProphecy->getParameter('kernel.debug')->willReturn(false);
 
+        $containerBuilderProphecy->getParameter('api_platform.eager_loading.enabled')->shouldBeCalled();
```
Comment (meyerbaptiste) [inline tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php]: Maybe you can make Coveralls and @Simperfit happy adding a test to fix this one: https://coveralls.io/builds/8650001/source?filename=src%2FBridge%2FSymfony%2FBundle%2FDependencyInjection%2FApiPlatformExtension.php#L127 😛

Comment (soyuka) [inline tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php]: Was doing it ^^, it's covered now ;)

Comment (soyuka) [inline tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php]: And this was bad, I had to access $config directly in the method ;)

**Comment by dunglas** [comment]
Thanks @soyuka

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 3]
```diff
@@ -65,6 +65,8 @@ public function load(array $configs, ContainerBuilder $container)
         $bundles = $container->getParameter('kernel.bundles');
 
         $this->registerMetadataConfiguration($container, $loader);
+        $this->registerDoctrineExtensionConfiguration($container, $config);
+        $this->registerSwaggerConfiguration($container, $config, $loader);
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Accidental duplicate call...

Comment (dunglas) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Good catch

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 4]
```diff
@@ -269,6 +271,19 @@ private function registerLoaders(ContainerBuilder $container, array $bundles)
     }
 
     /**
+     * Manipulate doctrine extension services according to the configuration.
+     *
+     * @param ContainerBuilder $container
+     * @param array            $config
+     */
+    private function registerDoctrineExtensionConfiguration(ContainerBuilder $container, array $config)
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: The function name reveals a problem with the configuration structure.

It should probably be something like:

```yaml
api_platform:
    doctrine:
        eager_loading: true
```

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php (hunk 3 above)
Comment (soyuka) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: Wow, good catch

**Pull request merged**